### PR TITLE
Zoe .d.ts libcheck

### PR DESCRIPTION
**Pull request by turadg**
refs: #9364

## Description

Follow-up to #9364, also doing the `skipLibCheck: false` for Zoe package and fixing up most of the issues. (Documenting the deferred).

### Upgrade Considerations

just types and tests. will not affect production but small impact on release process (cherry picking)

### CODE REVIEW

File: packages/zoe/test/unitTests/zcf/zcfSeat-exit.test.js [hunk 1]
```diff
@@ -67,8 +66,6 @@ test(`zoe - wrongly throw zcfSeat.exit()`, async t => {
 
   /** @type {OfferHandler} */
   const throwSeatFail = seat => {
-    // @ts-expect-error Linting correctly identifies that the argument to
```
Comment (turadg) [inline packages/zoe/test/unitTests/zcf/zcfSeat-exit.test.js:70]: `fail` actually takes `Details` which is `string | DetailsToken`.

File: packages/zoe/test/unitTests/zcf/zcfSeat-exit.test.js [hunk 2]
```diff
@@ -46,12 +46,11 @@ test(`zoe - wrongly throw zcfSeat.exit()`, async t => {
 
   // The contract uses the testJig so the contractFacet
   // is available here for testing purposes
-  /** @type {ContractFacet} */
+  /** @type {ZCF} */
   const zcf = testJig.zcf;
 
   /** @type {OfferHandler} */
   const throwSeatExit = seat => {
-    // @ts-expect-error Linting correctly identifies that exit takes no argument.
```
Comment (turadg) [inline packages/zoe/test/unitTests/zcf/zcfSeat-exit.test.js:54]: `exit` actually takes an optional `Completion` which is any non-thenable Passable

File: packages/vats/src/core/types-ambient.d.ts [hunk 3]
```diff
@@ -251,7 +251,14 @@ type WellKnownSpaces = {
   };
   instance: {
     produce: Record<WellKnownName['instance'], Producer<Instance>>;
-    consume: Record<WellKnownName['instance'], Promise<Instance>>;
+    consume: Record<WellKnownName['instance'], Promise<Instance>> & {
+      // not worth defining for all contracts. One here just for testing getPublicFacet(instance)
```
Comment (dckc) [inline packages/vats/src/core/types-ambient.d.ts]: bummer. oh well.

**Review by dckc — APPROVED** [review]

File: packages/vats/src/core/types-ambient.d.ts (hunk 3 above)
Comment (turadg) [reply packages/vats/src/core/types-ambient.d.ts]: happy to do it: 875f47e4a2

**Review by Chris-Hibbert — APPROVED** [review]
I have no objection, but that doesn't mean I understand what's changing here.

**Pull request merged**